terraform-provider-constellation/examples/full/gcp_cluster.tf

```diff
@@ -0,0 +1,113 @@
+terraform {
+  required_providers {
+    constellation = {
+      source  = "edgelesssys/constellation"
+      version = "X.Y.Z"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "3.6.0"
+    }
+  }
+}
+
+locals {
+  name                = "constell"
+  version             = "vX.Y.Z"
+  csp                 = "gcp"
+  attestation_variant = "gcp-sev-es"
+  region              = "europe-west3"
+  zone                = "europe-west3-b"
+  project_id          = "constellation-331613"
+
+  master_secret      = random_bytes.master_secret.hex
+  master_secret_salt = random_bytes.master_secret_salt.hex
+  measurement_salt   = random_bytes.measurement_salt.hex
+}
+
+resource "random_bytes" "master_secret" {
+  length = 32
+}
+
+resource "random_bytes" "master_secret_salt" {
+  length = 32
+}
+
+resource "random_bytes" "measurement_salt" {
+  length = 32
+}
+
+module "gcp_iam" {
+  // replace $VERSION with the Constellation version you want to use, e.g., v2.14.0
+  source             = "https://github.com/edgelesssys/constellation/releases/download/$VERSION/terraform-module.zip//terraform-module/iam/gcp"
+  project_id         = local.project_id
+  service_account_id = "${local.name}-test-sa"
+  zone               = local.zone
+  region             = local.region
+}
+
+module "gcp_infrastructure" {
+  // replace $VERSION with the Constellation version you want to use, e.g., v2.14.0
+  source = "https://github.com/edgelesssys/constellation/releases/download/$VERSION/terraform-module.zip//terraform-module/gcp"
+  name   = local.name
+  node_groups = {
+    control_plane_default = {
+      role          = "control-plane"
+      instance_type = "n2d-standard-4"
+      disk_size     = 30
+      disk_type     = "pd-ssd"
+      initial_count = 2
+      zone          = local.zone
+    },
+    worker_default = {
+      role          = "worker"
+      instance_type = "n2d-standard-4"
+      disk_size     = 30
+      disk_type     = "pd-ssd"
+      initial_count = 2
+      zone          = local.zone
+    }
+  }
+  image_id = data.constellation_image.bar.reference
+  debug    = false
+  zone     = local.zone
+  region   = local.region
+  project  = local.project_id
+}
+
+data "constellation_attestation" "foo" {
+  csp                 = local.csp
+  attestation_variant = local.attestation_variant
+  image_version       = local.version
+}
+
+data "constellation_image" "bar" {
+  csp                 = local.csp
+  attestation_variant = local.attestation_variant
+  image_version       = local.version
+}
+
+resource "constellation_cluster" "gcp_example" {
+  csp                                = local.csp
+  constellation_microservice_version = local.version
```

Comment:
I'd prefer staying expressive here, as this would allow users to easily override the value if desired without looking into the docs. I think the simplicity is fine here, due to the local variable allowing to easily set this in the "normal" case.